classes/submission/reviewAssignment/ReviewAssignmentDAO.inc.php

```diff
@@ -376,15 +375,21 @@ function insertObject($reviewAssignment) {
 		);
 
 		$reviewAssignment->setId($this->getInsertId());
-		return $reviewAssignment->getId();
+
+		// Update review stage status whenever a review assignment is changed
+		if ($result) {
+			$this->updateReviewRoundStatus($reviewAssignment);
+		}
+
+		return $reviewAssignment;
```

Comment:
Which `insert` functions are you referring to here? Something on the DAO? In this case, `ReviewAssignmentDAO::insertObject()` is called by `EditorAction::addReviewer()`. Are you suggesting that `EditorAction::addReviewer()` should be responsible for updating the object? (I can see either approach making sense. Just curious.)